src/EditorFeatures/Core/CommandHandlers/AbstractGoToCommandHandler.cs

```diff
@@ -2,132 +2,263 @@
 // The .NET Foundation licenses this file to you under the MIT license.
```

Comment:
this type is substantially rewritten.  i'll try to comment what is going on for clarity.